[julia-cp] iP

julia-cp opened the pull request:
### Blob

> Blob will help you organise your task.

- text based
- easy to learn

All you need is:

1. **download** it [here](https://github.com/julia-cp/ip/tree/master)
2. double click it
3. add your tasks
4. let it manage your tasks for you😉

and it's **FREE**


Features:

- [x] Managing tasks
- [x] Create To Dos
- [x] Create Deadlines
- [x] Create Events

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public static void main(String[] args) {
    new Duke("./data/storedTasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Level-0

Commit:
echo

Commit:
add, list

Commit:
adding Task class

Commit:
mark as done

Commit:
add ToDo, Event, Deadline classes

Commit:
automated text UI testing

Commit:
handle errors

Commit:
delete

Commit:
save

Ryo-Seah commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,128 @@
+            } else {
+                System.out.println("Sorry :(");
+                System.out.println("You need to use 'todo', 'deadline' or 'event' command to add a task.");
+                System.out.println("You can use 'list' to see all of your tasks.");
```
Good job Julia! Perhaps you could have used some more OOP principles in your code that was described in the tasks :)

Ryo-Seah commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
```
good naming conventions for all your methods!

Ryo-Seah reviewed (commented):
Overall good usage of naming conventions Julia! But I think you've noticed your Duke.java is getting very convoluted and hard to debug!

Commit:
save

Commit:
save

Commit:
Revert "delete"

This reverts commit 337630dcd462412ae95a81a1f2db84c4b652f6e8.

Commit:
Revert "Revert "delete""

This reverts commit 5256fa392080ce644e2d04ef3535b4febb02c7a5.

Commit:
Merge branch 'branch-Level-7'

sindhurajain commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,230 @@
+                case "E":
+                    task = new Event(taskInfo[2], taskInfo[3], taskInfo[4]);
+                    break;
+            }
```
Shouldn't there be no indentation for case clauses?

sindhurajain commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,230 @@
+                }
+            } else if (message.split(" ")[0].equalsIgnoreCase("event")) {
+                try {
+                    String task = message.split(" ", 2)[1];
```
I like how the if-else and try-catch statements are correctly formatted here

sindhurajain commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,44 @@
+        return this.getStatusIcon() + "/" + this.getDescription();
+    }
+
+    public void writeToFile(String filePath) throws IOException {
+        FileWriter fw = new FileWriter(filePath,true);
+        String fileString = fileString();
+        fw.write(fileString + System.lineSeparator());
+        fw.close();
+    }
```
Perhaps write header comments here as it's a public method

sindhurajain commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,230 @@
+                System.out.println();
+            } else if (message.split(" ")[0].equalsIgnoreCase("mark")) {
+                try {
+                    String number = message.split(" ")[1];
+                    int n = Integer.parseInt(number);
```
Perhaps more intuitive variable names here to differentiate between number and n?

sindhurajain commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,230 @@
+
+    private static void loadTasks(Scanner sc, ArrayList<Task> tasks) {
+        while (sc.hasNext()) {
+            String current = sc.nextLine();
```
Perhaps a more intuitive variable name here? Maybe currentLine instead of current?

sindhurajain reviewed (commented):
Overall, this looks good. The code is very readable and mostly follow the Java Coding Standard, only a few minor comments from me.

Commit:
date and time for deadline

Commit:
date and deadline for event

Commit:
add tasklist class

Commit:
storage class

Commit:
parser class

Commit:
Command class

Commit:
use more oop

Commit:
packages

Commit:
no message

Commit:
Merge branch 'master' into branch-A-Packages

Commit:
Merge branch 'add-gradle-support'

Commit:
add JUnit tests

Commit:
add JavaDoc comments

Commit:
Follow the coding standard

Commit:
Add FindCommand class

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Ui.java

Commit:
add GUI

Commit:
generate a new JAR file

Commit:
Assertion: add assertions to TaskList mark and unmark method

It is assumed that the mark and unmark method changes the status of the task.

An affirmation that this assumption holds is needed.

Let's update the TaskList mark and unmark method to include helpful assertions.

Commit:
Change the usage of Ui class according to its static methods.

Ui methods are associated with objects of Ui class.

Ui methods should be able to be called without the need for an object.

Let's,
* remove instantiation of Ui objects.
* associate the usage of the methods to Ui class.

Commit:
remove unused import statements

Commit:
Refactor a part of the execute method of the FindCommand class.

julia-cp committed:
Merge pull request #2 from julia-cp/branch-A-Assertions

Assertion: add assertions to TaskList mark and unmark method

julia-cp committed:
Merge branch 'master' into branch-A-CodeQuality

julia-cp committed:
Merge pull request #3 from julia-cp/branch-A-CodeQuality

Improve Code Quality

Commit:
Allow Event to be added in a non-specific order of from and to

Commit:
no message

Commit:
resize the window to be longer

Commit:
add a user guide

Commit:
Adding find to the Ui's showValidCommands method.

julia-cp committed:
update README.md file
